fix texture palette download (from mesa6: move grTexDownloadTable()
call from fxDDTexUseGlbPalette() to fxDDTexPalette().)  comment out
calls for downloading to TMU1 in case of GLIDE3 builds because code
is not using glide3 grTexDownloadTableExt() extension.

diff -urNp Mesa-3.4.2/src/FX/fxapi.c Mesa-3.4.2x/src/FX/fxapi.c
--- Mesa-3.4.2/src/FX/fxapi.c	2015-10-15 10:01:10.000000000 +0300
+++ Mesa-3.4.2x/src/FX/fxapi.c	2015-10-15 10:02:10.000000000 +0300
@@ -716,8 +716,10 @@ void GLAPIENTRY gl3DfxSetPaletteEXT(GLui
     fxMesa->haveGlobalPaletteTexture=1;
     
     FX_grTexDownloadTable(GR_TMU0,GR_TEXTABLE_PALETTE,(GuTexPalette *)pal);
+#if !defined(FX_GLIDE3)
     if (fxMesa->haveTwoTMUs)
     	 FX_grTexDownloadTable(GR_TMU1,GR_TEXTABLE_PALETTE,(GuTexPalette *)pal);
+#endif
   }
 }
 
diff -urNp Mesa-3.4.2/src/FX/fxddtex.c Mesa-3.4.2x/src/FX/fxddtex.c
--- Mesa-3.4.2/src/FX/fxddtex.c	2015-10-05 05:50:00.000000000 +0300
+++ Mesa-3.4.2x/src/FX/fxddtex.c	2015-10-15 10:02:10.000000000 +0300
@@ -603,6 +603,12 @@ void fxDDTexPalette(GLcontext *ctx, stru
     convertPalette(fxMesa->glbPalette.data, &ctx->Texture.Palette);
     fxMesa->new_state |= FX_NEW_TEXTURING;
     ctx->Driver.RenderStart = fxSetupFXUnits;
+
+    FX_grTexDownloadTable(GR_TMU0,GR_TEXTABLE_PALETTE, &(fxMesa->glbPalette));
+#if !defined(FX_GLIDE3)
+    if (fxMesa->haveTwoTMUs)
+       FX_grTexDownloadTable(GR_TMU1, GR_TEXTABLE_PALETTE, &(fxMesa->glbPalette));
+#endif
   }
 }
 
@@ -617,10 +623,6 @@ void fxDDTexUseGlbPalette(GLcontext *ctx
 
   if (state) {
     fxMesa->haveGlobalPaletteTexture = 1;
-
-    FX_grTexDownloadTable(GR_TMU0,GR_TEXTABLE_PALETTE, &(fxMesa->glbPalette));
-    if (fxMesa->haveTwoTMUs)
-       FX_grTexDownloadTable(GR_TMU1, GR_TEXTABLE_PALETTE, &(fxMesa->glbPalette));
   }
   else {
     fxMesa->haveGlobalPaletteTexture = 0;
diff -urNp Mesa-3.4.2/src/FX/fxsetup.c Mesa-3.4.2x/src/FX/fxsetup.c
--- Mesa-3.4.2/src/FX/fxsetup.c	2000-06-06 02:53:06.000000000 +0300
+++ Mesa-3.4.2x/src/FX/fxsetup.c	2015-10-15 10:02:10.000000000 +0300
@@ -351,7 +351,9 @@ static void fxSetupSingleTMU_NoLock(fxMe
 	  fprintf(stderr,"fxmesa: uploading texture palette\n");
        }
       FX_grTexDownloadTable_NoLock(GR_TMU0,GR_TEXTABLE_PALETTE,&(ti->palette));
+#if !defined(FX_GLIDE3)
       FX_grTexDownloadTable_NoLock(GR_TMU1,GR_TEXTABLE_PALETTE,&(ti->palette));
+#endif
     }
 
     FX_grTexClampMode_NoLock(GR_TMU0,ti->sClamp,ti->tClamp);
@@ -713,12 +715,14 @@ static void fxSetupDoubleTMU_NoLock(fxMe
        FX_grTexDownloadTable_NoLock(tmu0, GR_TEXTABLE_PALETTE, &(ti0->palette));
     }
 
+#if !defined(FX_GLIDE3)
     if (ti1->info.format==GR_TEXFMT_P_8) {
        if (MESA_VERBOSE&VERBOSE_DRIVER) {
 	  fprintf(stderr,"fxmesa: uploading texture palette TMU1\n");
        }
        FX_grTexDownloadTable_NoLock(tmu1, GR_TEXTABLE_PALETTE, &(ti1->palette));
     }
+#endif
   }
 
   FX_grTexSource_NoLock(tmu0, ti0->tm[tmu0]->startAddr, 
